components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/matchers/token/TokenParser.java

```diff
@@ -84,7 +85,16 @@ public TokenParser(final String templatePattern) {
 
     Map<String, String> values = new HashMap<>();
     for (int i = 0; i < matcher.groupCount(); i++) {
-      values.put(variables.get(i).getName(), matcher.group(i + 1));
+      if (values.containsKey(variables.get(i).getName())) {
+        if (!Objects.equals(values.get(variables.get(i).getName()), matcher.group(i + 1))) {
+          log.trace("Variable '{}' values mismatch: '{}' vs '{}'", variables.get(i).getName(),
+              values.get(variables.get(i).getName()), matcher.group(i + 1));
```

Comment:
probably look at making this more readable... a few locals could really help make this readable, and avoid duplication of work done to log.trace (which asis should be guarded, but if the code was simplified likely wouldn't).